### PR TITLE
fix: [CDS-47203]: Issues in Service dashboard

### PR DESCRIPTION
**Pull request by cjlee01**
### Summary

Issues in Service dashboard
- fixed issue when the hexagon is clicked a random diamond shaped box is being displayed near the value in Instances column (reference: video attachment)

#### Screenshots

NA
loading:
https://user-images.githubusercontent.com/92757601/205234224-e5c75bb3-c381-4fc6-9a57-db85bab370aa.mov

empty data:
https://user-images.githubusercontent.com/92757601/205234247-d82dd3e5-0e4f-467d-b5ae-fac10f613305.mov

error:
https://user-images.githubusercontent.com/92757601/205234250-27a44ec9-a7f5-49e8-90fd-3cbb3397875e.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

**Comment by ishantmehtaharness** [comment]
@Sagar-Rajput7 Could you also review

File: src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancePopover.tsx [hunk 1]
```diff
@@ -111,8 +113,28 @@ export const ActiveServiceInstancePopover: React.FC<ActiveServiceInstancePopover
...
+      <Card className={cx(css.activeServiceInstancePopover, css.spinner)}>
+        <Spinner />
+      </Card>
+    )
+  }
+
+  if (error) {
+    return (
+      <Card className={cx(css.activeServiceInstancePopover, css.spinner)}>
+        <Text className={css.errorText}>{error}</Text>
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancePopover.tsx:127]:
```suggestion
        <Text className={css.errorText}>{getErrorInfoFromErrorObject(error)}</Text>
```

Comment (Sagar-Rajput7) [reply src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancePopover.tsx:127]: This will break the whole page here.

**Review by Sagar-Rajput7 — CHANGES_REQUESTED** [review]

File: src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancePopover.tsx [hunk 2]
```diff
@@ -111,8 +113,28 @@ export const ActiveServiceInstancePopover: React.FC<ActiveServiceInstancePopover
...
+      </Card>
+    )
+  }
+
+  if (!data?.data?.instancesByBuildIdList?.[0]?.instances?.length) {
+    return (
+      <Card className={cx(css.activeServiceInstancePopover, css.spinner)}>
+        <Text>Instance data is empty.</Text>
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancePopover.tsx:135]: use getString here

**Review by ishantmehtaharness — APPROVED** [review]

File: src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancePopover.tsx [hunk 3]
```diff
@@ -111,8 +113,28 @@ export const ActiveServiceInstancePopover: React.FC<ActiveServiceInstancePopover
...
+    )
+  }
+
+  if (error) {
+    return (
+      <Card className={cx(css.activeServiceInstancePopover, css.spinner)}>
+        <Text className={css.errorText}>{getErrorInfoFromErrorObject(error)}</Text>
+      </Card>
+    )
+  }
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancePopover.tsx:130]: @cjlee01 Please see this. We should use getErrorInfoFromErrorObject while rendering an error object

Comment (ishantmehtaharness) [reply src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancePopover.tsx:130]: I also missed it

**Pull request merged**